rust/src/io/object_store.rs

```diff
@@ -283,11 +283,8 @@ impl ObjectStore {
 
         if !expanded_path.try_exists()? {
             std::fs::create_dir_all(expanded_path)?;
-        } else if !expanded_path.is_dir() {
-            return Err(Error::IO {
-                message: format!("{} is not a lance directory", str_path),
-            });
```

Comment:
This is necessary so we can re-use object store for other purposes. Otherwise it is tightly coupled to the lance directory.